test/test_torch.py

```diff
@@ -6972,67 +6972,70 @@ def test_logical_not_out(self, device, dtypes):
         np.logical_not(a.cpu().numpy(), out=out_np)
         self.assertEqual(out_np, out.to('cpu'))
 
-    def _test_logical(self, device, op, a_, b_, expected_res_):
-        for dtype in torch.testing.get_all_dtypes():
-            expected_res = torch.tensor(expected_res_, dtype=dtype, device=device)
-            a = torch.tensor(a_, dtype=dtype, device=device)
-            for other_dtype in torch.testing.get_all_dtypes():
-                b = torch.tensor(b_, dtype=other_dtype, device=device)
-
-                # TODO Remove this skipping after bfloat16 can be handled nicely with other dtypes.
-                # Skip only if either dtype or other_dtype is bfloat16, and promotion is unsupported.
-                if (dtype == torch.bfloat16) != (other_dtype == torch.bfloat16):
-                    non_bfloat16_dtype = dtype if dtype != torch.bfloat16 else other_dtype
-                    if non_bfloat16_dtype.is_complex or non_bfloat16_dtype == torch.float16:
-                        with self.assertRaises(RuntimeError):
-                            getattr(a, op)(b)
-                        continue
-
-                # Skip bfloat16 on CUDA. Remove this after bfloat16 is supported on CUDA.
-                # After type promotion of bfloat16 is supported, some bfloat16 logical operation will go through on
-                # CUDA as long as the two tensors are promoted to a supported type.
-                # TODO: Remove this once logical operators are improved to take care of bfloat16.
-                if self.device_type == 'cuda' and torch.bfloat16 in (dtype, other_dtype):
-                    if torch.promote_types(dtype, other_dtype) == torch.bfloat16:
-                        with self.assertRaises(RuntimeError):
-                            getattr(a, op)(b)
-                        continue
-
-                if dtype.is_complex or other_dtype.is_complex:
+    def _test_logical(self, device, dtype, op, a_, b_, expected_res_):
+        expected_res = torch.tensor(expected_res_, dtype=dtype, device=device)
+        a = torch.tensor(a_, dtype=dtype, device=device)
+        for other_dtype in torch.testing.get_all_dtypes():
+            b = torch.tensor(b_, dtype=other_dtype, device=device)
+
+            # TODO Remove this skipping after bfloat16 can be handled nicely with other dtypes.
+            # Skip only if either dtype or other_dtype is bfloat16, and promotion is unsupported.
+            if (dtype == torch.bfloat16) != (other_dtype == torch.bfloat16):
```

Comment:
Can this check be relaxed to test for bfloat16 x fp16 or fp16 x bfloat16?

Reply:
This part should be removed in the stacked PR, where bfloat16 support would be added for logical operators

Reply:
This is done two lines below